powerauth-credential-server-sample/src/main/java/io/getlime/security/powerauth/lib/credentials/controller/SMSAuthorizationController.java

```diff
@@ -0,0 +1,167 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.getlime.security.powerauth.lib.credentials.controller;
+
+import io.getlime.core.rest.model.base.request.ObjectRequest;
+import io.getlime.core.rest.model.base.response.ObjectResponse;
+import io.getlime.security.powerauth.lib.credentials.configuration.CredentialStoreConfiguration;
+import io.getlime.security.powerauth.lib.credentials.exception.SMSAuthorizationFailedException;
+import io.getlime.security.powerauth.lib.credentials.exception.SMSAuthorizationMessageInvalidException;
+import io.getlime.security.powerauth.lib.credentials.model.request.CreateSMSAuthorizationRequest;
+import io.getlime.security.powerauth.lib.credentials.model.request.VerifySMSAuthorizationRequest;
+import io.getlime.security.powerauth.lib.credentials.model.response.CreateSMSAuthorizationResponse;
+import io.getlime.security.powerauth.lib.credentials.repository.SMSAuthorizationRepository;
+import io.getlime.security.powerauth.lib.credentials.repository.model.entity.SMSAuthorizationEntity;
+import io.getlime.security.powerauth.lib.credentials.validation.CreateSMSAuthorizationRequestValidator;
+import org.joda.time.DateTime;
+import org.springframework.core.MethodParameter;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.BeanPropertyBindingResult;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.math.BigDecimal;
+import java.security.SecureRandom;
+import java.util.Date;
+import java.util.UUID;
+
+/**
+ * Controller class which handles SMS OTP authorization.
+ *
+ * @author Roman Strobl, roman.strobl@lime-company.eu
+ */
+@Controller
+public class SMSAuthorizationController {
+
+    private SMSAuthorizationRepository smsAuthorizationRepository;
+    private CredentialStoreConfiguration credentialStoreConfiguration;
+
+    public SMSAuthorizationController(SMSAuthorizationRepository smsAuthorizationRepository, CredentialStoreConfiguration credentialStoreConfiguration) {
+        this.smsAuthorizationRepository = smsAuthorizationRepository;
+        this.credentialStoreConfiguration = credentialStoreConfiguration;
+    }
+
+    /**
+     * Create a new SMS OTP authorization message.
+     *
+     * @param request Request data.
+     * @return Response with message ID.
+     */
+    @RequestMapping(value = "/sms/create", method = RequestMethod.POST)
+    public @ResponseBody
```

Comment:
Do not break lines between ResponseBody annotation and rest of the method signature, the code looks weird. I guess this is some IDEA formatting issue. You can either write:

```java
@RequestMapping(...)
@ResponseBody
public Object someMethod() {...}
```

... or ...

```java
@RequestMapping(...)
public @ResponseBody Object someMethod() {...}
```